src/Oscoin/Consensus/Nakamoto.hs

```diff
@@ -131,11 +131,18 @@ mineNakamoto
        , Hashable c (BlockHeader c (Sealed c PoW))
        )
     => Telemetry.Tracer m
-    -> (forall tx. [Block c tx (Sealed c PoW)] -> Telemetry.Traced Difficulty)
+    -> (forall tx. [Block c tx (Sealed c PoW)] -> Difficulty)
     -> Miner c PoW m
 mineNakamoto probed difiFn getBlocks unsealedBlock = do
     blks  <- getBlocks difficultyBlocks
-    diffy <- probed (difiFn blks)
+    let prevDifficulty = blockTargetDifficulty
+                       . blockHeader
+                       . NonEmpty.head
+                       . NonEmpty.fromList
```

Comment:
Do we actually need to convert to NonEmpty?

Reply:
one way or the the other we had to perform an unsafe operation, so I did stick to exactly what we do for the `prevDifficulty` inside `chainDifficulty`. Perhaps a better design would be to pass to the miner directly a NonEmpty?

Reply:
I mean, modulo turning this into a `Maybe` and then doing some unwrapping down the line, but if we have a bug here (i.e. an empty list), this will fail exactly one line below as well, as part of the call to `difiFn` -- imho the correct solution is to pass a `NonEmpty` in the first place, but this is outside the scope of this PR, methinks 😉